projects/plugins/jetpack/modules/search/class-jetpack-search-customberg.php

```diff
@@ -0,0 +1,145 @@
+require_once JETPACK__PLUGIN_DIR . '_inc/lib/admin-pages/class-jetpack-redux-state-helper.php';
+
+/**
+ * Responsible for adding a search customization interface to wp-admin.
+ *
+ * @package Automattic\Jetpack\Search
+ */
+class Jetpack_Search_Customberg {
+	/**
+	 * The singleton instance of this class.
+	 *
+	 * @var Jetpack_Search_Customberg
+	 */
+	protected static $instance;
+
+	/**
+	 * Get the singleton instance of the class.
+	 *
+	 * @return Jetpack_Search_Customberg
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new Jetpack_Search_Customberg();
+			self::$instance->init_hooks();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds action hooks.
+	 */
+	public function init_hooks() {
+		add_action( 'admin_menu', array( $this, 'add_wp_admin_page' ), 999 );
+	}
+
+	/**
+	 * Adds a wp-admin page without adding a sidebar submenu item.
+	 */
+	public function add_wp_admin_page() {
+		if ( ! $this->should_add_page() ) {
+			return;
+		}
+
+		// Intentionally omits adding a submenu via the first null argument.
+		$hook = add_submenu_page(
+			null,
+			__( 'Search Settings', 'jetpack' ),
+			__( 'Search', 'jetpack' ),
+			'manage_options',
+			'jetpack-search-customize',
+			array( $this, 'jetpack_search_admin_page' )
+		);
+		add_action( "admin_print_scripts-$hook", array( $this, 'load_admin_scripts' ) );
+		add_action( "admin_print_styles-$hook", array( $this, 'load_admin_styles' ) );
+	}
+
+	/**
+	 * Prints the dashboard container.
+	 *
+	 * @access public
+	 */
+	public function jetpack_search_admin_page() {
+		// TODO: Spin this function off into a static helper function in a helper class for code reuse.
+		$protocol   = is_ssl() ? 'https' : 'http';
+		$static_url = apply_filters( 'jetpack_static_url', "{$protocol}://en.wordpress.com/i/loading/loading-64.gif" );
+		?>
+			<div id="jp-search-customization" class="jp-search-customization-dashboard">
+				<div class="hide-if-no-js"><img class="jp-search-loader" width="32" height="32" alt="<?php esc_attr_e( 'Loading&hellip;', 'jetpack' ); ?>" src="<?php echo esc_url( $static_url ); ?>" /></div>
+				<div class="hide-if-js"><?php esc_html_e( 'Your Search customization page requires JavaScript to function properly.', 'jetpack' ); ?><div />
+			</div>
+		<?php
+	}
+
+	/**
+	 * Enqueue admin styles.
+	 */
+	public function load_admin_styles() {
+		\Jetpack_Admin_Page::load_wrapper_styles();
+
+		wp_enqueue_style(
+			'jp-search-customize',
+			plugins_url( '_inc/build/instant-search/jp-search-configure-main.bundle.css', JETPACK__PLUGIN_FILE ),
+			array(
+				'wp-components',
+				'wp-block-editor',
+			),
+			JETPACK__VERSION
+		);
+	}
+
+	/**
+	 * Enqueue admin scripts.
+	 */
+	public function load_admin_scripts() {
+		$script_deps_path    = JETPACK__PLUGIN_DIR . '_inc/build/instant-search/jp-search-configure-main.bundle.asset.php';
+		$script_dependencies = array( 'react', 'wp-i18n', 'wp-polyfill' );
```

Comment:
Is `wp-i18n` necessary here? Wouldn't it get added as soon as we start using the package in the bundle?

Reply:
This line serves as a backup list of dependencies in case the asset manifest doesn't generate as expected. I suppose throwing an error here might be better?

Reply:
Maybe? I suppose we're going to run into other issues down the road if that asset manifest isn't properly generated?

Reply:
For now, I'll go ahead and set the aforementioned line to a blank array. I don't think we've ever had a manifest generation issue so far, so let's cross that bridge when we get to it.